### PR TITLE
Fix 'save photo' exception

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

#1875 

#### What's this PR do?

Basically just ensures that if there isn't a photo selected we don't try to call the function to issue the request.

#### How should this be manually tested?

Open the photo upload dialog, and click 'save' without selecting a photo. You shouldn't see any errors in the JS console.

### CODE REVIEW

**Comment by amir-qayyum-khan** [comment]
@aliceriot is this open for review?

**Comment by aliceriot** [comment]
yes, I put the 'needs review' label on it.

**Comment by amir-qayyum-khan** [comment]
@aliceriot save is already unclickable 

screen shot 2016-11-29 at 1 41 34 am

**Comment by aliceriot** [comment]
yeah it is disabled (visually) but the `onclick` handler still fires when it is clicked. click the button with the JS console open, you'll see an exception (on `master`).

**Comment by amir-qayyum-khan** [comment]
good to go 👍

File: static/js/components/ProfileImageUploader.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ const ProfileImageUploader = ({
         type='button'
         className={`save-button ${photo ? 'primary-button' : 'secondary-button disabled'}`}
         key="save"
-        onClick={updateUserPhoto}>
+        onClick={photo ? updateUserPhoto : undefined}>
```
Comment (noisecapella) [inline static/js/components/ProfileImageUploader.js]: Wouldn't the `undefined` get executed as a function in this case?

Comment (noisecapella) [inline static/js/components/ProfileImageUploader.js]: I guess not actually, in HTML if the `onclick` is missing it's treated as if the function weren't there

Comment (aliceriot) [inline static/js/components/ProfileImageUploader.js]: I think that under the hood `props` are just objects, so `undefined` looks like the button didn't receive an `onClick` handler at all.

Comment (aliceriot) [inline static/js/components/ProfileImageUploader.js]: It doesn't cause an error or anything

**Pull request merged**